### PR TITLE
Bug 1863810: - panic: "invalid memory address or nil pointer dereference" while resyncing

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
Fix panic in `applyWebhook` if the server error occurs. Also, add tests to simulate server failure, and confirm the behavior is preserved.

### CODE REVIEW

**Comment by Danil-Grigorev** [comment]
PTAL @JoelSpeed @enxebre - return value fixed

File: pkg/operator/webhook_configuration.go [hunk 1]
```diff
@@ -174,6 +174,8 @@ func applyUnstructured(resourceClient dynamic.ResourceInterface, path string, re
 		actual, err := resourceClient.Create(context.TODO(), required, metav1.CreateOptions{})
```
Comment (enxebre) [inline pkg/operator/webhook_configuration.go:174]: for code bas consistency can we:
```
	if err != nil {
		if apierrors.IsNotFound(err) {
			actual, err := resourceClient.Create(context.TODO(), required, metav1.CreateOptions{})
			reportCreateEvent(recorder, required, err)
			return actual, true, err
		}
		return nil, false, err
	}


```

Comment (Danil-Grigorev) [reply pkg/operator/webhook_configuration.go]: @enxebre Applied the changes, PTAL

File: pkg/operator/webhook_configuration.go [hunk 2]
```diff
@@ -170,10 +170,13 @@ func applyUnstructured(resourceClient dynamic.ResourceInterface, path string, re
...
 	existing, err := resourceClient.Get(context.TODO(), required.GetName(), metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		actual, err := resourceClient.Create(context.TODO(), required, metav1.CreateOptions{})
-		reportCreateEvent(recorder, required, err)
-		return actual, true, err
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			actual, err := resourceClient.Create(context.TODO(), required, metav1.CreateOptions{})
+			reportCreateEvent(recorder, required, err)
+			return actual, true, err
```
Comment (JoelSpeed) [inline pkg/operator/webhook_configuration.go:177]: We should check the error here before we report the event as we have done on the Update later in this file

File: pkg/operator/sync_test.go [hunk 3]
```diff
@@ -311,17 +355,17 @@ func TestSyncMutatingWebhooks(t *testing.T) {
...
+	optr.testSyncWebhookConfiguration(t,
+		admissionregistrationv1.SchemeGroupVersion.WithResource("mutatingwebhookconfigurations"),
 		&unstructured.Unstructured{Object: configuration},
 		optr.syncMutatingWebhook,
 		stop,
 		optr.mutatingWebhookListerSynced, testCases)
 }
 
-func testSyncWebhookConfiguration(
+func (optr *Operator) testSyncWebhookConfiguration(
```
Comment (JoelSpeed) [inline pkg/operator/sync_test.go:366]: Why did this change to being a member of the `Operator`? Shouldn't the test code be completely separate from the `Operator`?

Comment (Danil-Grigorev) [reply pkg/operator/sync_test.go]: Did this change authomatically, fixed this to be an argument.

**Pull request merged**